operator: cleanup CRD registration

mhofstetter opened the pull request:
Cleanup CRD registration at operator startup by removing boilerplate functions and CRD v1beta1 support.

mhofstetter committed:
operator: cleanup CRD registration

Cleanup CRD registration at operator startup by removing boilerplate
functions.

Signed-off-by: Marco Hofstetter <marco.hofstetter@isovalent.com>

squeed commented on pkg/k8s/apis/cilium.io/client/register.go:
```diff
@@ -399,22 +240,15 @@ func createCNCCRD(clientset apiextensionsclient.Interface) error {
-	scopedLog := log.WithField("name", crdName)
+	scopedLog := log.WithField("name", crd.Name)
 
 	if !k8sversion.Capabilities().APIExtensionsV1CRD {
```
I'm pretty sure that this check can go away; can you confirm and, if reasonable, remove it too?

mhofstetter replied:
good point.

should be possible to remove since the oldest k8s release supported by stable cilium is k8s 1.16 (https://docs.cilium.io/en/stable/concepts/kubernetes/compatibility/#kubernetes-compatibility) and since k8s 1.16 CRD v1 is supported (https://kubernetes.io/docs/reference/using-api/deprecation-guide/#customresourcedefinition-v122).

going to remove this check and its logic

squeed reviewed (approved):
Looks good! If we can get rid of the v1beta1 CRD code, let's do that too.

mhofstetter committed:
operator: cleanup CRD v1beta1 support

Support for CRD v1beta1 has been removed, since this is no longer
necessary in all supported k8s versions.

Signed-off-by: Marco Hofstetter <marco.hofstetter@isovalent.com>
